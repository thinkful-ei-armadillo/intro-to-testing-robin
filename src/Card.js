import React from 'react';
import './Card.css';
import './List.js';

function Card(props){

  return(
    <div className='Card'>
      <h3>{props.title}</h3>
      <p>{props.content}</p>
      <button type="button" onClick={() => props.onDeleteClick(props.id)}>Delete</button>
    </div>
  );
}

export default Card;
