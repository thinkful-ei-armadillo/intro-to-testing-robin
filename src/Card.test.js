import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card';


// task - make a smoke test and snapshot test for Card.js

describe('<App />', () => {
  
  it('renders without crashing', () => {
      // Create a DOM element to render the component into
      const div = document.createElement('div');

      // Render the component
      // If something is wrong it will fail here
      ReactDOM.render(<Card />, div);

      // Clean up
      ReactDOM.unmountComponentAtNode(div);
  });

  it('renders without crashing', () => {
    // Render the component, as JSON
    const tree = renderer
      .create(<Card />)
      .toJSON();
    // Check whether it matches the previous snapshot
    // Stored in __snapshots__/App.test.js.snap
    expect(tree).toMatchSnapshot(); 
});

});