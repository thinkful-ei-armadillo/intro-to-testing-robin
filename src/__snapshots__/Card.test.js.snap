// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
<div
  className="Card"
>
  <h3 />
  <p />
</div>
`;
